### PR TITLE
Unify DDoc Makefile build variables

### PR DESCRIPTION
**Pull request by wilzbach**
Ideally this should cleanup our Ddoc variables mess. It's noteworthy that Druntime uses completely different ones, so the idea is to change the build files at Druntime afterwards.

If I did it right, _no_ changes should be detected by DAutoTest.

### CODE REVIEW

File: posix.mak [hunk 1]
```diff
@@ -10,6 +10,7 @@
 #
 
 include osmodel.mak
+PWD=$(realpath .)
```
Comment (wilzbach) [inline posix.mak]: The idea here to get rid of all the repeated `realpath` clutter

File: posix.mak [hunk 2]
```diff
@@ -107,6 +108,31 @@ else
 endif
 
 ################################################################################
+# Ddoc build variables
+################################################################################
+DDOC_VARS_STABLE=\
```
Comment (wilzbach) [inline posix.mak]: If we agree on [renaming _STABLE to _LATEST](https://github.com/dlang/dlang.org/pull/1689), this should be changed as well.
However without #1689 it the naming is `stable`...

File: posix.mak [hunk 3]
```diff
@@ -375,48 +401,39 @@ $(DMD_STABLE) : ${DMD_STABLE_DIR}
...
 ################################################################################
 # druntime, latest released build and current build
+# TODO: remove DOCDIR and DOCFMT once they have been removed at Druntime
 ################################################################################
 
 druntime-prerelease : ${DRUNTIME_DIR} $(DMD) $(STD_DDOC_PRE)
-	${MAKE} --directory=${DRUNTIME_DIR} -f posix.mak target doc \
+	${MAKE} --directory=${DRUNTIME_DIR} -f posix.mak target doc $(DDOC_VARS_HTML) \
 		DOCDIR=${DOC_OUTPUT_DIR}/phobos-prerelease \
 		DOCFMT="$(addprefix `pwd`/, $(STD_DDOC_PRE))"
```
Comment (wilzbach) [inline posix.mak]: https://github.com/dlang/druntime/pull/1834 changes `DOCDIR` and `DOCFMT` to their more common names. After the next release (2.074.2), these lines can be removed for all druntime doc targets.

File: posix.mak (hunk 1 above)
Comment (CyberShadow) [inline posix.mak]: `realpath` resolves symlinks and unnecessarily exposes implementation details in log files or even built files. We should get rid of it and just use `$(pwd)` instead.

**Comment by wilzbach** [comment]
> @`realpath` resolves symlinks and unnecessarily exposes implementation details in log files or even built files. We should get rid of it and just use `$(pwd)` instead.

You mean `$(shell pwd)`, right?
Done.

**Comment by CyberShadow** [comment]
Thanks!

>     Cloning into '/dev/shm/dtest/work/repo/dlang.org/../phobos-2.074.1'...

Looks like there's still a `readlink` somewhere...

**Comment by wilzbach** [comment]
> The term `DDOC_VARS_STABLE` doesn't relate to `DDOC_VARS_HTML` and `DDOC_VARS_VERBATIM`.
Following the existing naming schemas, it prolly should be named `DDOC_VARS_HTML_STABLE`,

https://github.com/dlang/dlang.org/pull/1714

**Comment by MartinNowak** [comment]
You also added a subtle breakage, now this makefiles requires any `DMD` or such variables to be absolute, while previously could deal well with relative paths.

File: posix.mak [hunk 4]
```diff
@@ -355,14 +381,14 @@ dlangspec.verbatim.txt : $(DMD) verbatim.ddoc dlangspec-consolidated.d
...
 ################################################################################
 
-../%-${LATEST} :
-	git clone -b v${LATEST} --depth=1 ${GIT_HOME}/$* $@
+$(PWD)/%-${LATEST} :
+	git clone -b v${LATEST} --depth=1 ${GIT_HOME}/$(notdir $*) $@
 
-../%-${DUB_VER} :
-	git clone --depth=1 -b v${DUB_VER} ${GIT_HOME}/$* $@
+$(PWD)/%-${DUB_VER} :
```
Comment (MartinNowak) [inline posix.mak]: Furthermore this broke the clone targets, they are located at `$(PWD)/../%-${LATEST}`.

**Pull request merged**